### PR TITLE
Issue#886 - Update issue imported from Gitpay after receive an invoice paid webhook

### LINKED ISSUE #886: Update issue imported from Gitpay after receive an invoice paid webhook
When a webhook send to update payment or invoice, we should put into the Gitpay flow:

- Send notifications
- Update Payment
- Display on the issue page
-  Be able to send transfers

### Webhook sent

```
{
  "object": {
    "id": ["in_test"]
    "object": "invoice",
    "amount_due": 6490,
    "charge": ["ch_test),
    "currency": "usd",
    "customer": ["cus_test),
    "default_payment_method": null,
    "default_source": null,
    "number": "USD-0016",
    "payment_intent": ["pi_test"]
    "period_end": 1636030341,
    "period_start": 1636030341,
    "status": "paid",
    "subscription": null,
    "total": 6490,
...
  }
}
```

### Why
We sometimes do payments from external sources, mainly using Stripe invoices from clients
We need to recognize this event to get the payment part of the test already charged for a task

### Level
Medium

### Current behavior
When a payment is done on Stripe, the task is not on Gitpay

### Expected behavior
When I use the metatag on Stripe about `order_id` I should create an order on Gitpay

### Useful links
  - Github API: https://docs.github.com/en/rest
  - Material-UI: https://material-ui.com/

### How to reproduce the user scenario
- Scenario 1
   - Run the project locally
   - Import an issue
   - Emulate a call to our webhooks with the object provided
   - Detect and change payment details

### What you will need to complete this issue
Stripe development keys

### Tips
- All the UI will use Material UI (https://material-ui.com/)
- Use our translation library Run `npm run translate` on the `/frontend` folder to sync translation before sending the PR

## EDIT

To clarify this, we will consider the following situation

We create orders from this screen that it goes directly on Stripe and can be paid
Screenshot 2022-03-24 at 10 33 05

This invoice is created here:
https://github.com/worknenjoy/gitpay/blob/42350a041d25fee37c6b7cec056f7db1d05b7744/frontend/src/components/task/assignment/AssignActions.js#L134

is an order of the type invoice-item

First of all, see the customer_id here:
https://github.com/worknenjoy/gitpay/blob/42350a041d25fee37c6b7cec056f7db1d05b7744/frontend/src/components/task/assignment/AssignActions.js#L143

If I'm creating, the invoice will be created, with my `customer_id` on Stripe
Now the first change we need to do:

_Subtask 1) Change this screen to have an email field, if I wish to pass another other than mine, because we need to pass another customer other than the one is sending_

We support clients paying directly by invoice and they don't need to have a Gitpay account

So we will pass the customer id and for and a user-id we have an issue

We have two options:
1) See the impact of this implementation, to create order without a user
2) Add to the user an activated flag and create new users with activated false and any other user true by default. This user will have a role of `funding`

_Subtask 2) From the Customer id input by user, get the customer on Stripe and try to fill only with e-mail:_
https://github.com/worknenjoy/gitpay/blob/42350a041d25fee37c6b7cec056f7db1d05b7744/frontend/src/components/task/assignment/AssignActions.js#L138

When this invoice is created the next step is to receive the webhook above when is paid and update this order.

This order created should be list as payments done in the task page

_Subtask 3) Save meta attributes on the invoice-item order with the task_id to be retrieved later by the webhook_

_Subtask 4) On the webhooks, you will receive the invoice paid webhook to change the status on Gitpay to paid
https://github.com/worknenjoy/gitpay/blob/e64d13d2d17381a617a867268e1e7348b0dec101/modules/app/controllers/webhook.js

### PR DESCRIPTION
**Pull request by VictorAlessander**
## Description

> Feature to support orders created and paid from funding accounts to be recognized and processed correctly by Gitpay.

> Plus: Fixing an issue with the route when trying to accept the assignment to work on the task causing route not found.

## Changes

- webhooks
- user model
- assign actions modal

## Issue

> Closes #886 

## Impacted Area

> Assign actions modal
> User model
> Webhook

## Steps to test

Steps needed to reproduce the scenario from this change to validate if the pull request solves this issue

- Create / log user
- Create Task
- Accept an interested user to work on the created task
- Create an order when assigning the user to the task
- Validate the webhook received from stripe

### CODE REVIEW

**Comment by VictorAlessander** [comment]
@alexanmtz 
Build failed because probably it's missing the seed-test command to run.

**Review by alexanmtz — CHANGES_REQUESTED** [review]

File: modules/app/controllers/webhook.js [hunk 1]
```diff
@@ -702,6 +703,33 @@ exports.updateWebhook = (req, res) => {
...
+                source_id: event.data.object.id[0]
+              },
+              returning: true
+            }
+          ).then(order => {
+            return res.json(req.body)
+          })
+        }).catch(e => {
+          return res.status(400).send(e)
+        })
```
[the remark this answers is not in the document]
Comment (VictorAlessander) [reply modules/app/controllers/webhook.js]: Done. Let me know if it's good now.

**Review by alexanmtz — CHANGES_REQUESTED** [review]
Great @VictorAlessander , now it's so much better now

We're still missing one of the requirements before we can move on:

_Subtask 1) Change this screen to have an email field, if I wish to pass another other than mine, because we need to pass another customer other than the one is sending_

we need to have an input with an email that when specified we create an order with that email rather than the current user, if is empty (or we can have a checkbox) we should get the current user

Can you do this adjustments?

**Comment by VictorAlessander** [comment]
> Great @VictorAlessander , now it's so much better now
> 
> We're still missing one of the requirements before we can move on:
> 
> _Subtask 1) Change this screen to have an email field, if I wish to pass another other than mine, because we need to pass another customer other than the one is sending_
> 
> we need to have an input with an email that when specified we create an order with that email rather than the current user, if is empty (or we can have a checkbox) we should get the current user
> 
> Can you do this adjustments?

Oh, sorry, I forgot that. Sure, going to implement.

**Comment by VictorAlessander** [comment]
@alexanmtz Ok, it's done. Let me know if is needed to do anything else.

**Pull request merged**